internal/generator/backend/template/representations/solution.go.tmpl

```diff
@@ -14,8 +14,10 @@ import (
 	{{ template "import_curve" . }}
 )
 
-// ErrUnsatisfiedConstraint can be generated when solving a R1CS
-var ErrUnsatisfiedConstraint = errors.New("constraint is not satisfied")
+// errUnsatisfiedConstraint can be generated when solving a R1CS
```

Comment:
Previously, as ErrUnsatisfiedConstraint was a instantiated type, it could be checked using errors.Is. Now, it is essentially a utility function to call fmt.Errorf. When looking at the places it is called, it makes difficult to understand where the error actually happened (grepping leads here, not to the actual place the error happened).
Maybe remove this function and construct new errors where necessary? If it is necessary to assert this error, then a conventional way would be to define this as a type (satisfying the error interface), and use errors.As for assertion.